CMakeLists.txt

```diff
@@ -244,16 +244,18 @@ endif ()
 # Add a section with the hash of the compiled machine code for integrity checks.
 # Only for official builds, because adding a section can be time consuming (rewrite of several GB).
 # And cross compiled binaries are not supported (since you cannot execute clickhouse hash-binary)
-if (OBJCOPY_PATH AND CLICKHOUSE_OFFICIAL_BUILD AND (NOT CMAKE_TOOLCHAIN_FILE OR CMAKE_TOOLCHAIN_FILE MATCHES "linux/toolchain-x86_64.cmake$"))
+if (CLICKHOUSE_OFFICIAL_BUILD AND (NOT CMAKE_TOOLCHAIN_FILE OR CMAKE_TOOLCHAIN_FILE MATCHES "linux/toolchain-x86_64.cmake$"))
+    message(STATUS "Official build: Will add a checksum hash to the clickhouse binary")
     set (USE_BINARY_HASH 1 CACHE STRING "Calculate binary hash and store it in the separate section")
+else ()
+    message(STATUS "Not an official build: Will not add a checksum hash to the clickhouse binary")
 endif ()
 
-# Allows to build stripped binary in a separate directory
-if (OBJCOPY_PATH AND STRIP_PATH)
```

Comment:
Removing it may break in funny places. This part has been adjusted very accurately for long time.

Reply:
You found the right reference already 😄

About: objcopy/strip: In general, I think it is worse to silently skip the initialization of split build (because we are on a hypothetical platform without objcopy) than it is to do the initialization and fail with a weird message later on. But that is all theory.

The main point of the commit is to prove what really happens in the CI builds. I looked at some generated binaries (stored at Amazon) and they had unexpected ELF sections.